### PR TITLE
gh-115264: Fix `test_functools` with `-00` mode

### PR DESCRIPTION
**Pull request by sobolevn**
Now the tests pass with both modes.

* Issue: gh-115264

### CODE REVIEW

**Comment by sobolevn** [comment]
Unrelated failure reported: https://github.com/python/cpython/issues/116455

File: Lib/test/test_functools.py [hunk 1]
```diff
@@ -2696,7 +2696,10 @@ def static_func(arg: int) -> str:
             A().static_func
         ):
             with self.subTest(meth=meth):
-                self.assertEqual(meth.__doc__, 'My function docstring')
+                self.assertEqual(meth.__doc__,
+                                 ('My function docstring'
+                                  if support.HAVE_DOCSTRINGS
+                                  else None))
```
Comment (AlexWaygood) [inline Lib/test/test_functools.py:2702]: Wow, I think this was one of the first tests I ever added to CPython! Sad that it had a bug 😆

File: Lib/test/test_functools.py [hunk 2]
```diff
@@ -3128,7 +3134,10 @@ def test_access_from_class(self):
         self.assertIsInstance(CachedCostItem.cost, py_functools.cached_property)
 
     def test_doc(self):
-        self.assertEqual(CachedCostItem.cost.__doc__, "The cost of the item.")
+        self.assertEqual(CachedCostItem.cost.__doc__,
+                         ("The cost of the item."
+                          if support.HAVE_DOCSTRINGS
+                          else None))
```
Comment (AlexWaygood) [inline Lib/test/test_functools.py:3140]: Or should this whole test just be skipped if we're in `-00` mode, maybe?

**Review by AlexWaygood — APPROVED** [review]

File: Lib/test/test_functools.py (hunk 2 above)
Comment (sobolevn) [reply Lib/test/test_functools.py:3140]: I think that checking that docstring is stripped in `-OO` is also helpful. So, I prefer to not skip it.

Comment (AlexWaygood) [reply Lib/test/test_functools.py:3140]: Sure. I don't have a strong opinion either way -- feel free to merge :)

**Pull request merged**